examples/fuse_examples/imaging/classification/knight/baseline/dataset.py

```diff
@@ -1,198 +1,264 @@
+from fuseimg.data.ops.aug.geometry import OpAugAffine2D, OpAugSqueeze3Dto2D, OpAugUnsqueeze3DFrom2D
+from fuseimg.data.ops.image_loader import OpLoadImage 
+from fuseimg.data.ops.color import OpClip, OpToRange
+from fuseimg.data.ops.shape_ops import OpRandomCrop
+
+import numpy as np
+from fuse.data.utils.sample import get_sample_id
+from typing import Hashable, List, Optional, Sequence, Tuple, Union
+from functools import partial
+import torch
+from torch import Tensor
+import pandas as pd
+from fuse.data.utils.collates import CollateDefault
+from fuse.data.utils.samplers import BatchSamplerDefault
+
+
+
+
+
+class OpKnightSampleIDDecode(OpBase):
+    '''
+    decodes sample id into image and segmentation filename
+    '''
+
+    def __call__(self, sample_dict: NDict) -> NDict:#, op_id: Optional[str]) -> NDict:
+        '''
+        
+        '''
+
+        sid = get_sample_id(sample_dict)
+        
+        img_filename_key = 'data.input.img_path'
+        sample_dict[img_filename_key] =   os.path.join(sid, 'imaging.nii.gz')
+
+        seg_filename_key = 'data.gt.seg_path'
+        sample_dict[seg_filename_key] = os.path.join(sid, 'aggregated_MAJ_seg.nii.gz')
+
+        
+        return sample_dict
+
+class OpClinicalLoad(OpBase):
+    def __init__(self, json_path: str):
+        super().__init__()
+        self.json_path = json_path
+
+    def __call__(self, sample_dict: NDict) -> NDict:
+        cols = ['case_id', 'age_at_nephrectomy', 'body_mass_index', 'gender', 'comorbidities', \
+                'aua_risk_group', 'smoking_history', 'radiographic_size', 'last_preop_egfr']
+
+        json_data = pd.read_json(self.json_path)[cols]
+        sid = sample_dict['data.input.img_path'][:-15]
+        row = json_data[json_data["case_id"]==sid].to_dict("records")[0]
+
+        row['gender'] = int(row['gender'].lower() == 'female') #female:1 | male:0
+        row["comorbidities"] = int(any(x for x in row["comorbidities"].values())) #if has any comorbidity it is set to 1
+        row['smoking_history'] = ['never_smoked','previous_smoker','current_smoker'].index(row['smoking_history'])
+        if row['last_preop_egfr'] is None or row['last_preop_egfr']['value'] is None:
+            row['last_preop_egfr'] = 77 # median value 
+        elif row['last_preop_egfr']['value'] in ('>=90', '>90'):
+            row['last_preop_egfr'] = 90
+        else:
+            row['last_preop_egfr'] = row['last_preop_egfr']['value']
+
+        if row['radiographic_size'] is None:
+            row['radiographic_size'] = 4.1 # this is the median value on the training set
+
+        sample_dict["data.gt.gt_global.task_1_label"] = int(row["aua_risk_group"] in ['high_risk', 'very_high_risk'])
+        sample_dict["data.gt.gt_global.task_2_label"] = ['benign','low_risk','intermediate_risk','high_risk', 'very_high_risk'].index(row["aua_risk_group"])
+
+        sample_dict["data.input.clinical"] = row
+
+        return sample_dict
+        
+class OpPrepare_Clinical(OpBase):
```

Comment:
Try to implement generic building blocks that can be reused in different datasets.
Use OpLookup (you might need minor modification) or implement a new op to handle missing values.
Use OpToRange to move it a range between 0 and 1.
Use OpToOneHot to convert to one-hot vector.
And OpConcat to concatenate.